Flagsmith.Engine/Segment/Evaluator.cs

```diff
@@ -276,45 +273,6 @@ private static FlagResult<FeatureMetadataT> GetFlagResult<_, FeatureMetadataT>(E
 
     public static class Evaluator
     {
-        public static Hashing Hashing = new Hashing();
-
-        public static List<SegmentModel> GetIdentitySegments(EnvironmentModel environmentModel, IdentityModel identity, List<TraitModel> overrideTraits)
-            => environmentModel.Project.Segments.Where(s => EvaluateIdentityInSegment(identity, s, overrideTraits)).ToList();
-
-        public static bool EvaluateIdentityInSegment(IdentityModel identity, SegmentModel segment, List<TraitModel> overrideTraits)
-        {
-            var traits = overrideTraits?.Any() == true ? overrideTraits : identity.IdentityTraits;
-            var identityHashKey = identity.DjangoId == null ? identity.CompositeKey : identity.DjangoId.ToString();
-            return segment.Rules?.Any() == true && segment.Rules.All(rule => TraitsMatchSegmentRule(traits, rule, segment.Id.ToString(), identityHashKey));
-        }
-
-        static bool TraitsMatchSegmentRule(List<TraitModel> identityTraits, SegmentRuleModel rule, string segmentId, string identityId)
-        {
-            var matchesConditions = !rule.Conditions.Any() || rule.MatchingFunction(rule.Conditions.Select(c =>
-                    TraitsMatchSegmentCondition(identityTraits, c, segmentId, identityId)).ToList()
-            );
-            return matchesConditions && (rule.Rules?.All(r => TraitsMatchSegmentRule(identityTraits, r, segmentId, identityId)) ?? true);
-        }
-
-        static bool TraitsMatchSegmentCondition(List<TraitModel> identityTraits, SegmentConditionModel condition, string segmentId, string identityId)
-        {
-            if (condition.Operator == Constants.PercentageSplit)
-                return Hashing.GetHashedPercentageForObjectIds(new List<string>() { segmentId, identityId }) <= float.Parse(condition.Value);
-
-            var trait = identityTraits?.FirstOrDefault(t => t.TraitKey == condition.Property);
-
-            if (condition.Operator == Constants.IsSet)
-            {
-                return trait != null;
-            }
-            else if (condition.Operator == Constants.IsNotSet)
-            {
-                return trait == null;
-            }
-
-            return trait != null && MatchesTraitValue(trait.TraitValue, condition);
-        }
-
         public static bool MatchesTraitValue(object traitValue, SegmentConditionModel condition)
```

Comment:
I expect this function to change to:

```suggestion
        public static bool MatchesContextValue(object contextValue, Condition condition)
```

Reply:
Good catch. Fixed in 5a0fe23f4c41461acb5620f7230b5707083227c5.

Reply:
Not quite — it still accepts the old engine `SegmentConditionModel` class. I expect it to accept `Condition`.

Reply:
Attempted with 1734178856088312e1d14a85796ed7f2d345daa7.